syntax = "proto2";

package auth_protocol;

enum Action {
    SIGN_UP = 0;
    LOGIN = 1;
}

enum Error {
    REPEATED_USERNAME = 0;
    INVALID_CREDENTIALS = 1;
    INTERNAL_SERVER_ERROR = 2;
}

message Request {
    required int32 length = 1;
    required Action action = 2;
    required string username = 3;
    required string plaintextPassword = 4;
}

message ResponseError {
    required int32 length = 1;
    required Action action = 2;
    required string username = 3;
    required string error = 4;
    required bool vilchis = 5;
}

message ResponseOK {
    required int32 length = 1;
    required Action action = 2;
    required string username = 3;
}
